src/polkit/polkitunixprocess.c

```diff
@@ -165,6 +165,7 @@ struct _PolkitUnixProcess
   guint ctty;
   gboolean pidfd_is_safe;
   GArray *gids;
+  gboolean constructed;
```

Comment:
This is pointless and confusing. The object is either created with a pid or a pidfd, not both. Please drop this commit, and the next one. The first two commits are good.

Reply:
All `G_PARAM_CONSTRUCT` properties means all of them, including default values. If you don't specify a "pid" during construction GLib will act as if "pid" is set to 0 (vice versa with "pidfd" except it's -1). This case must be handled.

Reply:
It is already handled. This serves no purpose, please drop this commit.

Reply:
Only for the case where `pid` is set before `pidfd` during construction. Since the order in which properties are set during construction is entirely arbitrary and unspecified, there is no guarantee that this ordering behavior will be the same in future GLib versions, which may set `pid` after `pidfd` and break construction of `PolkitUnixProcess`es from pidfds.

Reply:
Once again, the actual reported issue is fixed by the first commit, the rest is working as intended. Please drop this commit, the first two are good and can be merged. Thanks.